Wikipedia/Code/DiffListChangeItemViewModel.swift

```diff
@@ -222,6 +225,69 @@ final class DiffListChangeItemViewModel {
+                    modifiedAccessibilityText = (modifiedAccessibilityText as NSString).replacingCharacters(in: nsRange, with: String.localizedStringWithFormat(diffChangeRemoval, originalChange))
+                }
+            }
+        case .moveSource:
+            if let moveDirection = moveInfo?.linkDirection, let moveDistance = moveInfo?.moveDistance {
+
+                let diffMoveSourceLinesUp = WMFLocalizedString("diff-change-paragraph-lines-moved-up", value: "Paragraph moved up {{PLURAL:%1$d|0=0 lines|%1$d line|%1$d lines}}. Go to new location.", comment: "Accessibility text announced in VoiceOver to indicate content has moved. %1$d is replaced by the number of lines.")
+                let diffMoveSourceLinesDown = WMFLocalizedString("diff-change-paragraph-lines-moved-down", value: "Paragraph moved down {{PLURAL:%1$d|0=0 lines|%1$d line|%1$d lines}}. Go to new location.", comment: "Accessibility text announced in VoiceOver to indicate content has moved. %1$d is replaced by the number of lines.")
+                let diffMoveSourceSectionUp = WMFLocalizedString("diff-change-paragraph-sections-moved-up", value: "Paragraph moved up {{PLURAL:%1$d|0=0 sections|%1$d section|%1$d sections}}. Go to new location.", comment: "Accessibility text announced in VoiceOver to indicate content has moved. %1$d is replaced by the number of sections.")
+                let diffMoveSourceSectionsDown = WMFLocalizedString("diff-change-paragraph-sections-moved-down", value: "Paragraph moved down {{PLURAL:%1$d|0=0 sections|%1$d section|%1$d sections}}. Go to new location.", comment: "Accessibility text announced in VoiceOver to indicate content has moved. %1$d is replaced by the number of sections.")
+
+                switch moveDistance {
+                case .line(let amount):
+                    if moveDirection == .up {
+                        modifiedAccessibilityText = String.localizedStringWithFormat(diffMoveSourceLinesUp, amount)
+                    } else {
+                        modifiedAccessibilityText = String.localizedStringWithFormat(diffMoveSourceLinesDown, amount)
+                    }
+                case .section(let amount):
+                    if moveDirection == .up {
+                        modifiedAccessibilityText = String.localizedStringWithFormat(diffMoveSourceSectionUp, amount)
+                    } else {
+                        modifiedAccessibilityText = String.localizedStringWithFormat(diffMoveSourceSectionsDown, amount)
+                    }
+                }
+
+                modifiedAccessibilityText += text
+            }
+        case .moveDestination:
+            let diffParagraphMoved = WMFLocalizedString("diff-change-paragraph-moved", value: "Paragraph moved. Go to previous location.", comment: "Accessibility text announced in VoiceOver to indicate content has moved.")
+            modifiedAccessibilityText = diffParagraphMoved + text
```

Comment:
I notice this "text" can have some integrated deletions and additions. Maybe you could run this text through [this logic](https://github.com/wikimedia/wikipedia-ios/pull/4771/files#diff-3d5f1b06bbd8cbd1d63ac39e6999b4267584e6d5a1918f039e1bd5810635caa8R240-R255) from above to get those "removal" / "addition" bits?

![IMG_0336](https://github.com/wikimedia/wikipedia-ios/assets/3620196/84363111-b108-402a-a376-5e43bca66c83)

Reply:
Good catch – updated all the diff change types except add/removal lines to use that logic.